tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/impl/MetricVariable.java

```
@@ -0,0 +1,125 @@
+   * Creates a Metric Variable
+   *
+   * @param tf the TensorFlow Ops
+   * @param variable the variable
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variable
+   */
+  public MetricVariable(Ops tf, Variable<U> variable, long seed, Class<U> type) {
+    this(tf, variable, null, seed, type);
+  }
+
+  /**
+   * Creates a Metric Variable
+   *
+   * @param tf the TensorFlow Ops
+   * @param variable the variable
+   * @param initializer the initializer for the variable, if null, then the default for floating
+   *     point types is {@link org.tensorflow.framework.initializers.Glorot} with distribution
+   *     {@link org.tensorflow.framework.initializers.VarianceScaling.Distribution#UNIFORM}, for
+   *     other types the default initializer is {@link org.tensorflow.framework.initializers.Zeros}
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variable
+   */
+  @SuppressWarnings("unchecked")
+  public MetricVariable(
+      Ops tf, Variable<U> variable, Initializer<U> initializer, long seed, Class<U> type) {
+    this.tf = tf;
+    this.variable = variable;
+
+    if (initializer == null) {
+      if (TFloating.class.isAssignableFrom(type)) {
+        //noinspection RedundantCast
+        this.initializer =
+            (Initializer<U>) new Glorot<>(tf, VarianceScaling.Distribution.UNIFORM, seed);
+      } else if (TIntegral.class.isAssignableFrom(type)) {
+        this.initializer = new Zeros<>(tf);
+      } else {
+        throw new IllegalArgumentException(
+            String.format(
+                "An initializer for variable %s of type %s is required",
+                variable.toString(), type.getSimpleName()));
```

an `IllegalArgumentException` should be thrown if this is a user exception, while here it is a missing implementation. Either we should throw `UnsupportedOperationException` or we should change the error message for something like "Type %s is not a supported for metric variables"

---

Changed the error message as suggested, and fixed the @throws in Java Doc.